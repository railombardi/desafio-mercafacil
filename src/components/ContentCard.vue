<template>
  <div class="card">
    <span class="card__title">{{ title }}</span>
    <span class="card__subtitle">{{ subtitle }}</span>
    <span class="card__extra">{{ extra }}</span>
  </div>
</template>

<script>
export default {
  props: {
    title: {
      type: String,
      default: "",
    },
    subtitle: {
      type: String,
      default: "",
    },
    extra: {
      type: String,
      default: "",
    },
  },
};
</script>

<style lang="scss" scoped>
.card {
  display: flex;
  flex-direction: column;
  align-items: center;
  background: #fafafa;
  box-shadow: 0px 1px 5px rgba(0, 0, 0, 0.2), 0px 3px 4px rgba(0, 0, 0, 0.12),
    0px 2px 4px rgba(0, 0, 0, 0.14);
  border-radius: 4px;
  width: 312px;
  height: 128px;
  padding: 25px 16px;
  transition: all 0.3s ease;
  cursor: pointer;
  @include tablet() {
    width: 240px;
    &:hover {
      width: 260px;
    }
  }
  &__title {
    font-weight: 500;
    font-size: 20px;
    line-height: 30px;
    text-align: center;
  }
  &__subtitle {
    font-weight: 400;
    font-size: 14px;
    line-height: 21px;
    color: $text-placeholder;
  }
  &__extra {
    font-style: normal;
    font-weight: 700;
    font-size: 14px;
    line-height: 21px;
    color: $text-placeholder;
  }
}
</style>
