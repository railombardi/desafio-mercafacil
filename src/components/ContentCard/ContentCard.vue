<template>
  <div class="card">
    <span class="card__title">{{ title }}</span>
    <span class="card__subtitle">{{ subtitle }}</span>
    <span class="card__extra">{{ extra }}</span>
  </div>
</template>

<script>
export default {
  props: {
    /**
     * Card title
     */
    title: {
      type: String,
      default: "",
    },
    /**
     * Card subtitle
     */
    subtitle: {
      type: String,
      default: "",
    },
    /**
     * Card field for additional information
     */
    extra: {
      type: String,
      default: "",
    },
  },
};
</script>

<style lang="scss" scoped>
.card {
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  background: #fafafa;
  box-shadow: 0px 1px 5px rgba(0, 0, 0, 0.2), 0px 3px 4px rgba(0, 0, 0, 0.12),
    0px 2px 4px rgba(0, 0, 0, 0.14);
  border-radius: 4px;
  width: 312px;
  height: 128px;
  padding: 2px 16px;
  transition: all 0.3s ease;
  cursor: pointer;
  text-align: center;
  @include tablet() {
    width: 240px;
    &:hover {
      width: 260px;
    }
  }
  &__title {
    font-weight: 500;
    font-size: 20px;
    line-height: 30px;
    text-align: center;
  }
  &__subtitle {
    font-weight: 400;
    font-size: 14px;
    line-height: 21px;
    color: $text-placeholder;
  }
  &__extra {
    font-style: normal;
    font-weight: 700;
    font-size: 14px;
    line-height: 21px;
    color: $text-placeholder;
  }
  &__title,
  &__subtitle,
  &__extra {
    overflow: hidden;
    text-overflow: ellipsis;
    max-width: 100%;
    white-space: nowrap;
  }
  &__title,
  &__subtitle {
    @supports (-webkit-line-clamp: 2) {
      max-width: 100%;
      max-height: 60px;
      overflow: hidden;
      text-overflow: ellipsis;
      white-space: initial;
      display: -webkit-box;
      -webkit-line-clamp: 2;
      -webkit-box-orient: vertical;
      word-wrap: break-word;
    }
  }
}
</style>
