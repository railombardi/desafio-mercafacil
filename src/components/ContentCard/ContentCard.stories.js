import ContentCard from "./ContentCard.vue";
export default {
  title: "Components/ContentCard",
  component: ContentCard,
  argTypes: {},
};

const Template = (args, { argTypes }) => ({
  props: Object.keys(argTypes),
  components: {
    ContentCard,
  },
  template: `<div style="display: flex; justify-content: center">
              <content-card :title="title" :subtitle="subtitle" :extra="extra" />
             </div>`,
});

export const Default = Template.bind({});
Default.args = {
  title: "Card title",
  subtitle: "Card subtitle",
  extra: "Card field for additional information",
};
export const LocationCard = Template.bind({});
LocationCard.args = {
  title: "Earth (C-137)",
  subtitle: "Planet",
};
export const EpisodeCard = Template.bind({});
EpisodeCard.args = {
  title: "Pilot",
  subtitle: "December 2, 2013",
  extra: "SE01E01",
};
