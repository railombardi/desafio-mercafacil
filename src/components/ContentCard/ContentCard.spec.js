import { shallowMount, createLocalVue } from "@vue/test-utils";
import ContentCard from "./ContentCard.vue";

const localVue = createLocalVue();

let wrapper = shallowMount(ContentCard, {
  localVue,
});

describe("ContentCard.vue", () => {
  it("Snapshot test", () => {
    expect(wrapper.html()).toMatchSnapshot();
  });
  describe("[PROPS] =>", () => {
    it("When the prop title receives a value, the content is displayed in the target span", async () => {
      const text = "Test title prop";
      await wrapper.setProps({ title: text });
      const span = wrapper.find(".card__title");
      expect(span.text()).toBe(text);
    });
    it("When the prop subtitle receives a value, the content is displayed in the target span", async () => {
      const text = "Test subtitle prop";
      await wrapper.setProps({ subtitle: text });
      const span = wrapper.find(".card__subtitle");
      expect(span.text()).toBe(text);
    });
    it("When the prop extra receives a value, the content is displayed in the target span", async () => {
      const text = "Test extra prop";
      await wrapper.setProps({ extra: text });
      const span = wrapper.find(".card__extra");
      expect(span.text()).toBe(text);
    });
  });
});
