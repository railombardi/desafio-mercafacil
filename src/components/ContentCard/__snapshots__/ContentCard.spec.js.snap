// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ContentCard.vue Snapshot test 1`] = `<div class="card"><span class="card__title"></span> <span class="card__subtitle"></span> <span class="card__extra"></span></div>`;
